nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -140,9 +146,15 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
     }
 
     protected NuxeoPrincipal getAuthor(DocumentModel docModel) {
-        String[] contributors = (String[]) docModel.getProperty("dublincore", "contributors");
         UserManager userManager = Framework.getService(UserManager.class);
-        return userManager.getPrincipal(contributors[0]);
+        try {
+            // this is more reliable in case of a comment
+            Property authorProp = docModel.getProperty(COMMENT_AUTHOR);
+            return userManager.getPrincipal((String) authorProp.getValue());
+        } catch (PropertyNotFoundException e) {
```

Comment:
can you give us, an example on which cases we can get the `PropertyNotFoundException` i think we can find another way instead of rely on the exception to get the second information for example i was thinking of things like: 

```
 protected NuxeoPrincipal getAuthor(DocumentModel docModel) {
        String author = null;
        if (docModel.hasSchema(COMMENT_SCHEMA)) {
            // means annotation / comment
            author = (String) docModel.getPropertyValue(COMMENT_AUTHOR);
        }

        if (StringUtils.isEmpty(author)) {
            String[] contributors = (String[]) docModel.getProperty("dublincore", "contributors");
            // to be more robust we can check if the  the array is emtpy ...and log ...but i think we will always have an element 
            author = contributors[0];
        }

        return Framework.getService(UserManager.class).getPrincipal(author);
    }
```

Reply:
This works fine, I replaced the second `if` by an `else if` though, because `COMMENT_AUTHOR` prevails as we discussed f2f